### PR TITLE
[gn] mac entitlement config when dart sdk is non prebuilt 

### PR DESCRIPTION
**Pull request by XilaiZhang**
a followup from https://github.com/flutter/engine/pull/34605, to add mac entitlement config when dart sdk is not prebuilt.

### CODE REVIEW

File: build/archives/BUILD.gn [hunk 1]
```diff
@@ -172,17 +172,24 @@ if (build_engine_artifacts && !flutter_prebuilt_dart_sdk) {
...
-    }
     files = [
       {
         source = rebase_path("$root_build_dir/dart-sdk")
         destination = "dart-sdk"
       },
     ]
+    if (is_mac) {
+      # Mac artifacts sometimes use mac and sometimes darwin. Standardizing the
+      # names will require changes in the list of artifacts the tool is downloading.
```
Comment (zanderso) [inline build/archives/BUILD.gn:183]: It sounds like this comment should be a TODO with a link to a GitHub issue.

Comment (XilaiZhang) [reply build/archives/BUILD.gn:183]: Umm I am not familiar with the context of this part, I just copied this paragraph.  Maybe the author @godofredoc  could help take a look, thank you

**Comment by zanderso** [comment]
From PR review triage: @XilaiZhang I think it's probably safe to convert it to a TODO and file an issue and we can follow-up with @godofredoc afterwords.

**Comment by godofredoc** [comment]
> From PR review triage: @XilaiZhang I think it's probably safe to convert it to a TODO and file an issue and we can follow-up with @godofredoc afterwords.

There is a bug already for this, let me find it

**Comment by godofredoc** [comment]
> > From PR review triage: @XilaiZhang I think it's probably safe to convert it to a TODO and file an issue and we can follow-up with @godofredoc afterwords.
> 
> There is a bug already for this, let me find it

https://github.com/flutter/flutter/issues/105351 is where we are documenting all those details.

**Comment by XilaiZhang** [comment]
Great, thanks everyone! I will remove this block of text

**Review by zanderso — APPROVED** [review]

**Pull request merged**